### ISSUE TITLE
`cargo clippy` not obeying `[lints.clippy]` from `Cargo.toml`, but only for some lints

### ISSUE THREAD

**Issue by Kmeakin** [rust-lang/rust-clippy#11237]
### Summary

Clippy does not obey the `lint_name = "allow"` key under `[lints.clippy]` from `Cargo.toml` for certain lints, but does work as expected for others. The lints have to be silenced with `#[allow(...)]` instead.

In the example below, `must_use_candidate` obeys the `[lints.clippy]` settings, but `wildcard_imports` doesn't.

Other lints that don't obey `[lints.clippy]` that I have encountered:
* `too_many_lines`
* `option_if_let_else`

### Reproducer

`Cargo.toml`:
```toml
[package]
name = "clippy-test"
version = "0.1.0"
edition = "2021"

[lints.clippy]
pedantic = "warn"

must_use_candidate = "allow"
wildcard_imports = "allow"
```

`.cargo/config.toml`:
```toml
[unstable]
lints = true

```

`src/lib.rs`:
```rust
mod foo {
    pub struct A;
}

// "usage of wildcard import" reported
use foo::*;

// "this function could have a `#[must_use]` attribute" not reported
pub fn a() -> A {
    A
}
```

### Version

```text
rustc 1.73.0-nightly (864bdf784 2023-07-25)
binary: rustc
commit-hash: 864bdf7843e1ceabc824ed86d97006acad6af643
commit-date: 2023-07-25
host: x86_64-unknown-linux-gnu
release: 1.73.0-nightly
LLVM version: 16.0.5
```
```


### Additional Labels

_No response_

**Comment by Alexendoo** [comment]
Looks like a cargo issue, it sorts the flags before passing them to clippy but the order matters

```
'--allow=clippy::wildcard_imports' '--warn=clippy::pedantic' '--allow=clippy::must_use_candidate'
```

**Comment by Alexendoo** [comment]
Closing in favour of https://github.com/rust-lang/cargo/issues/12918

**Comment by kristof-mattei** [comment]
As posted here https://github.com/rust-lang/cargo/issues/12918#issuecomment-1818071667

The behavior here seems different than rustc. rustc seems to consolidate the lints, apply groups and then the individual settings, whereas Clippy seems to go by the order of the params (and thus the priority).

Wouldn't it be better to match rustc's behavior?

**Comment by Alexendoo** [comment]
Clippy has no knowledge of the lint table, cargo turns it into `--allow`/`--warn`/etc flags the same as it would pass to `rustc` in `cargo check`/`cargo build`
